examples/deepResearch/firecrawlProvider.tsx

```diff
@@ -0,0 +1,46 @@
+import FirecrawlApp, { FirecrawlAppConfig } from "@mendable/firecrawl-js";
+import { gsx } from "gensx";
+
+// Create a context
+export const FirecrawlContext = gsx.createContext<{
+  client?: FirecrawlApp;
+}>({});
+
+// Create the provider
+export const FirecrawlProvider = gsx.Component<FirecrawlAppConfig, never>(
```

Comment:
We should consider turning this into a package at some point.

Reply:
agreed, i already actually had to duplicate this code from the providers example so even more reason to turn it into a package.

Reply:
We should have a group discussion (that we should have had earlier probably) about the tradeoffs/ergonomics of individual packages (`@gensx/firecrawl`) vs an umbrella package (eg `@gensx/services`). There are maintenance burden tradeoffs I think.

Reply:
I definitely lean towards individual packages even considering the tooling burden, but happy to have that discussion sooner rather than later.